### ISSUE TITLE
[Bug]: 

### ISSUE THREAD

**Issue by TaylorHuston** [jekyll/jekyll#9233]
### Operating System

Ubuntu 22.04

### Ruby Version

ruby 3.2.0 (2022-12-25 revision a528908271) [x86_64-linux]

### Jekyll Version

jekyll 3.9.2

### Expected Behavior

I am able to serve a basic Jekyll site with 
`bundle exec jekyll serve`

### Current Behavior

"Liquid Exception: undefined method `untaint' for "Welcome to Jekyll!":String in /_layouts/post.html
jekyll 3.9.2 | Error:  undefined method `untaint' for "Welcome to Jekyll!":String
/home/thuston/.rbenv/versions/3.2.0/lib/ruby/gems/3.2.0/gems/liquid-4.0.3/lib/liquid/standardfilters.rb:42:in `escape': undefined method `untaint' for "Welcome to Jekyll!":String (NoMethodError)"

I tried to update my blog after ignoring it for some time and was unable to do so, every time I tried to Jekyll Serve I would get the error "Liquid Exception: undefined method `tainted?' for "Code":String in /_layouts/page.html"

Wasn't able to find an immediate fix so I removed Ubuntu from WSL, installed 22.04, re-installed Jekyll and everything, got the same error. So then I created a new Jekyll site with just `jekyll new test`, tried to serve that, and for the error listed above.

So fresh Linux install, fresh Jekyll install, fresh site. Any ideas?

**Comment by ashmaroli** [comment]
The only solution currently is to uninstall Ruby 3.2. Install Ruby 3.1.2 (or older upto Ruby 2.7.x).

**Comment by gingerbeardman** [comment]
How do I install Ruby 3.1.2, with chruby? 

The install guide https://jekyllrb.com/docs/installation/macos/ installs Ruby 3.2.0 but expects to have installed 3.1.2

**Comment by Sandy-Kim** [comment]
Ruby 3.1.2 is not stable. You should install Ruby 3.1.3. [Check stable release](https://www.ruby-lang.org/en/downloads/)
```
ruby-install ruby 3.1.3
```

If you get surf same problem, you can see [this documents](https://docs.github.com/en/pages/setting-up-a-github-pages-site-with-jekyll/testing-your-github-pages-site-locally-with-jekyll)
Look at the Note number 4.

Note: If you've installed Ruby 3.0 or later (which you may have if you installed the default version via Homebrew), you might get an error at this step. That's because these versions of Ruby no longer come with webrick installed.

To fix the error, try running `bundle add webrick`, then re-running bundle exec jekyll serve.

**Comment by ashmaroli** [comment]
@Sandy-Kim I would not say Ruby 3.1.2 is not stable.
Sure, Ruby 3.1.3 is the current latest in Ruby 3.1 series, but our CI couldn't install `liquid-c` on Ruby 3.1.3 and therefore cannot recommend it with certainty.

**Comment by gingerbeardman** [comment]
> You should install Ruby 3.1.3

@Sandy-Kim I had to use that anyway because 3.1.2 doesn't work with M1 Apple Silicon.

> our CI couldn't install liquid-c on Ruby 3.1.3 

@ashmaroli I also use liquid-c with Jekyll, and it works for me just fine locally and on Netlify. Let me know if I can provide any further information.

**Comment by ashmaroli** [comment]
@gingerbeardman Thanks for letting me know you could successfully install and use `liquid-c` on macOS. Any idea if anyone was likewise successful on Linux / Ubuntu ..?

**Comment by gingerbeardman** [comment]
@ashmaroli yesterday I did a Jekyll (re)install using the official guide if that helps.

**Comment by ashmaroli** [comment]
@gingerbeardman Was that on Linux or macOS?

**Comment by gingerbeardman** [comment]
macOS, but my point was that reinstalling fixed my been setup and liquid-c is working. Maybe the same will work for you on your platform?

**Comment by eloydegen** [comment]
Just wanted to add that I have the same issue with GitLab pages, while it worked fine before: https://gitlab.com/retrowrt/retrowrt.gitlab.io/-/pipelines/735450438

**Comment by jhss** [comment]
I changed the ruby version to 3.1.3 in deploy yml file, and it worked.

**Comment by eloydegen** [comment]
@jhss That fixed the issue for me as well, with [this commit](https://gitlab.com/retrowrt/retrowrt.gitlab.io/-/commit/081fb97fa58264610954cbd387c5186a37a06188). So it seems to be some regression with Ruby 3.2.0. 

@Sandy-Kim Was that page you linked changed recently? Because it says that Ruby 3.2.0 is a stable release.

**Comment by ashmaroli** [comment]
@eloydegen it's not a regression but an expected effect of a planned removal of certain methods from the Ruby language core.

**Comment by e0218a** [comment]
is the exception here being thrown from liquid or jekyll?

which one is calling the method ```untaint```

**Comment by note4** [comment]
> https ://gitlab.com/retrowrt/retrowrt.gitlab.io/-/pipelines/735450438

你可以修改你的 .gitlab-ci.yml 文件

image: ruby:2.7

**Comment by ashmaroli** [comment]
To Everyone intending to use Jekyll on Ruby 3.2, we have decided to patch the issue at our end with #9248. 
Please point your Gemfile to the pull request and let us know if your build still fails:

```ruby
# in your Gemfile
gem "jekyll", github: "jekyll/jekyll", ref: "refs/pull/9248/head"
```

**Comment by ashmaroli** [comment]
Resolved with release of [Liquid 4.0.4](https://rubygems.org/gems/liquid/versions/4.0.4)
